cmd/validate.go

```diff
@@ -249,8 +255,12 @@ func processEntry(ctx context.Context, pw progress.Writer, wp cmdutil.Pool, e pl
 			// Group children by ones that look "similar", and select one from each group to test.
 			groups := make(map[criteria][]plugin.Entry)
 			for _, entry := range entries {
-				crit := newCriteria(entry)
-				groups[crit] = append(groups[crit], entry)
+				ccrit := newCriteria(entry)
+				// If we have a TypeID, only explore children if they are different from the parent.
+				// This prevents simple recursion like volume directories containing more dirs.
+				if ccrit.typeID != "" && ccrit != crit {
```

Comment:
Yes, I think this is a good heuristic. Only thing I'd add is to maybe print the TypeID in the console? That way, we can see what kind of thing is being tested.

Reply:
Oh also, do you think it's worth mentioning this heuristic in the help text (since it is not strictly correct?) I lean towards "No until a user files an issue about it."

Reply:
Same.

Reply:
I'll work TypeID into the output if present.

Reply:
Done.